sources/web/datalab/polymer/modules/GapiManager.ts

```diff
@@ -114,50 +143,61 @@ class GapiManager {
       projectId,
       tableId,
     };
-    return GapiManager._loadBigQuery().then(() => gapi.client.bigquery.tables.get(request));
+    return GapiManager._loadBigQuery()
+      .then(() => gapi.client.bigquery.tables.get(request));
   }
 
-  private static _initClient(signInChangedCallback: (signedIn: boolean) => void) {
-    // Initialize the client with API key and People API, and initialize OAuth with an
-    // OAuth 2.0 client ID and scopes (space delimited string) to request access.
-    gapi.client.init({
+  public static listenForSignInChanges(signInChangedCallback: (signedIn: boolean) => void):
+      Promise<void> {
+    return this.loadGapi()
+      .then(() => {
+        // If auth status is already know, initialize the callback now
+        if (gapi && gapi.auth2) {
```

Comment:
Yes, this code used to live outside the promise so the check was necessary. Done.